staging_vespalib/src/vespa/vespalib/util/adaptive_sequenced_executor.cpp

```diff
@@ -256,6 +256,11 @@ AdaptiveSequencedExecutor::~AdaptiveSequencedExecutor()
     assert(_worker_stack.empty());
 }
 
+ISequencedTaskExecutor::ExecutorId
+AdaptiveSequencedExecutor::getExecutorId(uint64_t component) const {
```

Comment:
this bypasses a lot of magic implemented in the 'interface'; is that intentional? if yes, move magic to the other subclass. If no, move magic to utility function that can be called from this one.

Reply:
Yes, that is intentional. I planned to move the magic out of the interface in a separate PR. I can put it in this one if you prefer that.